### ISSUE TITLE
speed

### ISSUE THREAD

**Issue by mosjoker** [MaybeShewill-CV/lanenet-lane-detection#18]
hi,i tested the model  on GTX-1080 , but i found it's too slow approaching 1 fps.
"The deep neural network inference part can achieve around a 50fps"--can u tell me the reason?
thank you,@MaybeShewill-CV

**Comment by MaybeShewill-CV** [comment]
@mosjoker Please check if the image feed pip line is the bottleneck

**Comment by mosjoker** [comment]
oh,i solved it.but i found another problem.The clustering part takes more time than the network forward,but in the papar was less than one-half of the network forward.
Have you tried other resolutions? I mean the network input.or weights quantified(float---int)?

**Comment by derekwong66** [comment]
@mosjoker May I ask how did you solve the latency problem? I encountered the same issue. Loading image and post-processing are fast, roughly 0.005S. However, prediction is really slow, which is about 0.5s per image. I am using GTX-1080. Thanks!

**Comment by MaybeShewill-CV** [comment]
@derekwong66 You test a single image or test a batch of images

**Comment by derekwong66** [comment]
@MaybeShewill-CV thanks your reply. 
I tried both. roughly the same prediction time.
$python tools/test_lanenet.py --is_batch True --batch_size 2 --save_dir data/tusimple_test_image/results/ --weights_path /home/zihang/project/lanenet-lane-detection/model/tusimple_v2/tusimple_lanenet_vgg_2018-10-19-13-33-56.ckpt-200000 --image_path data/tusimple_test_image/ --use_gpu 1
I1114 10:44:05.004429 27989 test_lanenet.py:196] [Epoch:153] 预处理2张图像, 共耗时: 0.04910s, 平均每张耗时: 0.02455
I1114 10:44:06.042906 27989 test_lanenet.py:203] [Epoch:153] 预测2张图像车道线, 共耗时: 1.03837s, 平均每张耗时: 0.51918s
I1114 10:44:06.161880 27989 test_lanenet.py:233] [Epoch:153] 进行2张图像车道线聚类, 共耗时: 0.01696s, 平均每张耗时: 0.00848

$python tools/test_lanenet.py --is_batch False --batch_size 1 --save_dir data/tusimple_test_image/results/ --weights_path /home/zihang/project/lanenet-lane-detection/model/tusimple_v2/tusimple_lanenet_vgg_2018-10-19-13-33-56.ckpt-200000 --image_path data/tusimple_test_image/0.png --use_gpu 1
I1114 10:46:24.991937 28538 tf_logging.py:115] Restoring parameters from /home/zihang/project/lanenet-lane-detection/model/tusimple_v2/tusimple_lanenet_vgg_2018-10-19-13-33-56.ckpt-200000
I1114 10:46:25.548634 28538 test_lanenet.py:115] 单张图像车道线预测耗时: 0.50682s

**Comment by MaybeShewill-CV** [comment]
@derekwong66 If you test a batch of image you can adjust the batch size to 32 for example:)

**Comment by derekwong66** [comment]
@MaybeShewill-CV  I tried 32 and 8. seems not different.
batch 32:
I1114 11:16:56.971650 29045 test_lanenet.py:196] [Epoch:6] 预处理32张图像, 共耗时: 0.49305s, 平均每张耗时: 0.01541
I1114 11:17:12.432114 29045 test_lanenet.py:203] [Epoch:6] 预测32张图像车道线, 共耗时: 15.46036s, 平均每张耗时: 0.48314s
I1114 11:17:13.926787 29045 test_lanenet.py:233] [Epoch:6] 进行32张图像车道线聚类, 共耗时: 0.41856s, 平均每张耗时: 0.01308

batch 8:
I1114 11:19:20.011592 29256 test_lanenet.py:186] [Epoch:9] 开始图像读取和预处理...
I1114 11:19:20.177321 29256 test_lanenet.py:196] [Epoch:9] 预处理8张图像, 共耗时: 0.16567s, 平均每张耗时: 0.02071
I1114 11:19:24.116097 29256 test_lanenet.py:203] [Epoch:9] 预测8张图像车道线, 共耗时: 3.93867s, 平均每张耗时: 0.49233s
I1114 11:19:24.546752 29256 test_lanenet.py:233] [Epoch:9] 进行8张图像车道线聚类, 共耗时: 0.08357s, 平均每张耗时: 0.01045

**Comment by derekwong66** [comment]
@MaybeShewill-CV I solved this problem by reinstalling tensorflow-gpu 1.10
Thanks for your kindly reply.
This is the latest result. In total, lane detection can run at around 25FPS
I1114 11:35:42.557834 29981 test_lanenet.py:186] [Epoch:5] 开始图像读取和预处理...
I1114 11:35:42.728412 29981 test_lanenet.py:196] [Epoch:5] 预处理8张图像, 共耗时: 0.17050s, 平均每张耗时: 0.02131
I1114 11:35:42.836155 29981 test_lanenet.py:203] [Epoch:5] 预测8张图像车道线, 共耗时: 0.10762s, 平均每张耗时: 0.01345s
I1114 11:35:43.260143 29981 test_lanenet.py:233] [Epoch:5] 进行8张图像车道线聚类, 共耗时: 0.09188s, 平均每张耗时: 0.01149

**Comment by MaybeShewill-CV** [comment]
@derekwong66 Welcome:)

**Comment by derekwong66** [comment]
Hi, @MaybeShewill-CV 

If I run is_batch True batch size = 1 
$python tools/test_lanenet.py --is_batch True --batch_size 1 --save_dir data/tusimple_test_image/results/ --weights_path /home/zihang/project/lanenet-lane-detection/model/tusimple_v2/tusimple_lanenet_vgg_2018-10-19-13-33-56.ckpt-200000 --image_path data/tusimple_test_image/ --use_gpu 1
the prediction time will be 0.01628s
I1116 11:59:02.694891 3362 test_lanenet.py:196] [Epoch:28] 预处理1张图像, 共耗时: 0.01645s, 平均每张耗时: 0.01645
I1116 11:59:02.711267 3362 test_lanenet.py:203] [Epoch:28] 预测1张图像车道线, 共耗时: 0.01628s, 平均每张耗时: 0.01628s
I1116 11:59:02.758955 3362 test_lanenet.py:233] [Epoch:28] 进行1张图像车道线聚类, 共耗时: 0.01296s, 平均每张耗时: 0.01296
If I run is_batch False batch size = 1
The prediction time will be 0.8s
I1116 12:11:54.235382 3533 test_lanenet.py:115] 单张图像车道线预测耗时: 0.82736s
I run one image for both, but the prediction time is different. Can you tell me what is the reason?

**Comment by derekwong66** [comment]
@MaybeShewill-CV 
you're right! I will set a warm-up process.
Thanks!
